pkg/components/state/pluggable.go

```diff
@@ -265,7 +268,7 @@ func (ss *grpcStateStore) BulkGet(ctx context.Context, req []state.GetRequest) (
 			ContentType: strNilIfEmpty(resp.ContentType),
 		}
 	}
-	return bulkGetResponse.Got, items, nil
```

Comment:
(Aside from the fact that it was an odd API design choice; should have been an optional interface method instead). All state store components now implement `BulkGet`, either natively or through the default bulk store in components-contrib. So this isn't needed.

The "looping" across all keys has been moved to components-contrib, since before it was implemented 3 times in the runtime, not consistently (causing bugs like those fixed in #6134). One of the 3 instances had a TODO (with your name :) ) to move the code to components-contrib.